packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -117,22 +175,45 @@ class CupertinoSwitch extends StatefulWidget {
   /// If null and [applyTheme] is false, defaults to [CupertinoColors.systemGreen]
   /// in accordance to native iOS behavior. Otherwise, defaults to
   /// [CupertinoThemeData.primaryColor].
+  ///
+  /// See also:
+  ///
+  ///  * [trackColor], the color to use for the track when the switch is off.
   final Color? activeColor;
 
-
   /// The color to use for the track when the switch is off.
   ///
   /// Defaults to [CupertinoColors.secondarySystemFill] when null.
+  ///
+  /// See also:
+  ///
+  ///  * [activeColor], the color to use for the track when the switch is on.
   final Color? trackColor;
 
-  /// The color to use for the thumb of the switch.
+  /// The color to use for the thumb when the switch is on.
   ///
-  /// Defaults to [CupertinoColors.white] when null.
+  /// Defaults to [CupertinoColors.white] when null or if the color provided has
+  /// no opacity.
```

Comment:
Can you clarify this? "no opacity" sounds like an opacity of zero to me, but it looks like it means an opacity < 1.0? And it doesn't look like it always defaults to `CupertinoColors.white` in that case, but to a blur against the scaffold color in the theme?

Reply:
Yes, that's true. The thumb color is blended to the theme surface color, so when the opacity is 0 it just uses that color instead. The color was white by default so I made the mistake of generalizing.